agent/taskresource/envFiles/envfile.go

```diff
@@ -0,0 +1,198 @@
+package envFiles
+
+import (
+	"sync"
+	"time"
+
+	apicontainer "github.com/aws/amazon-ecs-agent/agent/api/container"
+	"github.com/aws/amazon-ecs-agent/agent/credentials"
+	resourcestatus "github.com/aws/amazon-ecs-agent/agent/taskresource/status"
+	"github.com/aws/amazon-ecs-agent/agent/s3/factory"
+	"github.com/cihub/seelog"
+	"github.com/aws/amazon-ecs-agent/agent/s3"
+	"path/filepath"
+	"strings"
+	"github.com/pkg/errors"
+	"github.com/aws/amazon-ecs-agent/agent/utils/oswrapper"
+
+	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
+	"github.com/aws/amazon-ecs-agent/agent/taskresource"
+	"github.com/aws/amazon-ecs-agent/agent/api/task/status"
+)
+
+const (
+	envFileDirPath = "envfiles"
+	envTempFilePrefix = "tmp_env"
+
+	s3DownloadTimeout = 30 * time.Second
+)
+
+// represents envfile as a task resource
+// these env files are retrieved from s3
+type EnvironmentFileResource struct {
+	cluster   string
+	taskARN   string
+	createdAt time.Time
+	region    string
+	resourceDir string // path to store env var files
```

Comment:
We are not capturing the "type" value (currently only s3) in this object, and implementation including error message assumes s3. We don't have to over engineer ant this point, but it'd be a good data to capture and at least make it available for future iteration.

Reply:
good point, will add

Reply:
on second thought, "type" wasn't included because we have a list of the environment files - each object in the list could be of a different type (if we ever support anything besides s3). 
 concerns/thoughts?

Reply:
can't we store the type per file? we are accepting a list of env files, are you modeling the entire list as 1 task resource?

Reply:
type IS stored per file - each EnvironmentFile object has a Value and Type. Unless you mean each file should be a separate resource? So each object in the list is its own resource? I've currently modeled the entire list as 1 task resource because that's how it's defined from the payload.